src/Numerics/Mesh/Interpolation.jl

```diff
@@ -737,7 +737,7 @@ struct InterpolationCubedSphere{
         qm1 = poly_order + 1
         toler1 = FT(eps(FT) * vert_range[1] * 2.0) # tolerance for unwarp function
         toler2 = FT(eps(FT) * 4.0)                 # tolerance
-        toler3 = FT(eps(FT) * vert_range[1] * 10.0) # tolerance for Newton-Raphson
+        toler3 = 10^5 * eps(FT) / vert_range[1] + 10 * eps(FT) * vert_range[1] # tolerance for Newton-Raphson
```

Comment:
I divided the tolerance offset by `vert_range[1]` (the radius) so that the offset disappears for large planets and we roughly recover the original tolerance.

Reply:
A hardwired 10^5 here seems weird. Any number that is not O(1) should be related to something physical (I don't know what it is in the context here).

Reply:
True. I can change to `sqrt(eps(FT))`:
```julia
julia> 10^5*eps(Float64)
2.220446049250313e-11

julia> sqrt(eps(Float64))
1.4901161193847656e-8
```

Reply:
Maybe better yet to understand where these tolerances come from. Is `vert_range` dimensional or non-dimensional? If it is dimensional, l. 740 is not dimensionally consistent.

Reply:
I don't know what these are used for either, but my guess if you want something related to `eps(maximum(vert_range)))`

Reply:
I believe this is code that @skandalaCLIMA wrote.

Reply:
The tolerance needed to be set in relation to planet radius as there is precision loss when dealing with large numbers(planet radius). This was especially important for Float32 runs. But, we shouldn’t need to hardwire tolerance numbers though.

Reply:
The tolerance needed to be set in relation to planet radius as there is precision loss when dealing with large numbers(planet radius). This was especially important for Float32 runs. But, we shouldn’t need to hardwire tolerance numbers though.

Reply:
Can you suggest something that is dimensionally correct (i.e., does not have one term of units 1/m, added to a term of units m, as in the current l. 740?

Reply:
The dimension of toler3, I believe, is meters. Easiest change would be to reduce the tolerance (multiplication by a factor of 10 or 100) or increase the number of Newton-Raphson iterations. Let me try to replicate this.

Reply:
I ran a test (test/Numerics/Mesh/interpolation.jl) with _planet_radius = 1.0 and domain_height/_planet_radius = 30000/6.371e6 = 0.0047088. **The Float64 ran fine with an L_infinity max domain error of ~8E-8.** However, Float32 produced L_infinity max domain error of 0.0104.

Reply:
@charleskawczynski : Please advise on the exact parameters (planet radius and domain height) used for the test run.

Reply:
I'll try to set up an MWE

Reply:
@skandalaCLIMA, here's an MWE (or Minimum Breaking Example? is MBE a thing?):
```julia
using ClimateMachine
ClimateMachine.init()
using ClimateMachine.Atmos
using ClimateMachine.ConfigTypes
using ClimateMachine.Diagnostics
using ClimateMachine.Mesh.Interpolation
using CLIMAParameters
using CLIMAParameters.Planet: planet_radius
import CLIMAParameters
struct EarthParameterSet <: AbstractEarthParameterSet end
const param_set = EarthParameterSet()
CLIMAParameters.Planet.planet_radius(::EarthParameterSet) = 1
function main()
    FT,n_horz,n_vert,poly_order = Float64,12,6,3
    model = AtmosModel{FT}(AtmosGCMConfigType, param_set; init_state_prognostic = x->x)
    domain_height = FT(30e3)
    _planet_radius = FT(planet_radius(param_set))
    driver_config = ClimateMachine.AtmosGCMConfiguration(
        "SmallPlanet", poly_order, (n_horz, n_vert),
        domain_height, param_set, x->x; model = model,
    )
    info = driver_config.config_info
    boundaries = [ FT(-90) FT(-180) _planet_radius
        FT(90) FT(180) FT(_planet_radius + info.domain_height)]
    resolution = (FT(2), FT(2), FT(1000)) # in (deg, deg, m)
    interpol = ClimateMachine.InterpolationConfiguration(
        driver_config,
        boundaries,
        resolution,
    )
    return nothing
end
main()
```

Reply:
@charleskawczynski : The grid in the example above seems to be covering 30,000 m in the z direction for a 1 m radius planet. Can you please confirm if this is intended?

Reply:
Yes, I realize that this is for more pencil-shaped atmospheric columns, which I suppose we should avoid. That said, it still fails at 10x aspect ratio:
```julia
CLIMAParameters.Planet.planet_radius(::EarthParameterSet) = 3000
...
    domain_height = FT(3000*10)
```

Reply:
And 4x aspect ratio at `planet_radius = 10^4`:
```julia
CLIMAParameters.Planet.planet_radius(::EarthParameterSet) = 10^4
...
    domain_height = FT(10^4*4)
```

Reply:
Would the whole problem be fixed if a `vert_range[end]` was used. Then you would be using the largest number instead of the smaller.

Reply:
More thoroughly, passing in `domain_height`, and running

```julia
results = Dict()
for pr in 10.0 .^(0:6)
    CLIMAParameters.Planet.planet_radius(::EarthParameterSet) = 10^4
    for ar in 10.0 .^(-10:3:10)
        domain_height = ar*planet_radius(param_set)
        results[pr, ar] = false
        try
            main(domain_height)
            results[pr, ar] = true
        catch
        end
    end
end
```
Yields
```julia
julia> count(values(results))/length(values(results))
0.5714285714285714
```

Maybe we can add this as a test as
```julia
@test all(values(results))
```
?

Reply:
The parameter toler3 is used in the non-linear solver, that essentially estimates inner coordinates from unwarped exterior coordinates. Toler3 measures how good the result is from expected.

30000x or 15000x could result in mesh that are agressively streched. One way would be to contruct the tolerance using both interior and exterior radius.

A better way would be to have a local toler3 for each spectral element, or one for each direction of the spectral element.

However, I believe the domain height/radius ratios we use are tiny (<<1), I it is not clear if the additional effort is justified.



> Yes, I realize that this is for more pencil-shaped atmospheric columns, which I suppose we should avoid. That said, it still fails at 5x aspect ratio:
> 
> ```julia
> CLIMAParameters.Planet.planet_radius(::EarthParameterSet) = 3000
> ...
>     domain_height = FT(3000*5)
> ```

The aspect ratio here would be 15000 right?

Reply:
More thoroughly, passing in `domain_height`, and running

```julia
results = Dict()
for pr in 10.0 .^(0:7)
    CLIMAParameters.Planet.planet_radius(::EarthParameterSet) = 10^4
    for ar in 10.0 .^(-1:0.2:1)
        domain_height = ar*planet_radius(param_set)
        results[pr, ar] = false
        try
            main(domain_height)
            results[pr, ar] = true
        catch
        end
    end
end
```
Yields
```julia
julia> count(values(results))/length(values(results))
0.7272727272727273
```

Maybe we can add this as a test as
```julia
@test all(values(results))
```
?

Reply:
> The aspect ratio here would be 15000 right?
I'm calling `domain_height/planet_radius` the aspect ratio, in which case it would be 5.

Reply:
I'll split off the other part of this PR, since that seems more straightforward.

Reply:
The current aspect ratio, used for earth simulations, is 0.0047088. 0.727 seems to be a factor of ~182x higher. Do we have an estimate of what aspect ratios we will be using. I can make toler3 cell specific, if we intend to use more agressive aspect ratios frequently in future.

Reply:
> > The aspect ratio here would be 15000 right?
> 
> I'm calling `domain_height/planet_radius` the aspect ratio, in which case it would be 5.

Yes. I agree. I missed that the radius was also changed here.